### PR TITLE
Serix struct tag changes

### PR DESCRIPTION
**Pull request by muXxer**
- Remove the serix "position" from the field tag
- Use the first position as the fieldKey (former `mapKey`)
- If fieldKey is empty, use camelCase with some exceptions (`ID` => `Id`, `URL` => `Url`, etc...)
- Added `maxByteSize`
- Rename some tags
   - `lengthPrefixType` => `lenPrefix`
   - `nest`=> `inlined`

### CODE REVIEW

File: ds/set_impl.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func (s *set[ElementType]) apply(mutations SetMutations[ElementType]) (appliedMu
 //
 //nolint:tagliatelle // heck knows why this linter fails here
 type readableSet[T comparable] struct {
-	*orderedmap.OrderedMap[T, types.Empty] `serix:"0"`
+	*orderedmap.OrderedMap[T, types.Empty] `serix:","`
```
Comment (piotrm50) [inline ds/set_impl.go]: Do we always need at least the comma?

File: serializer/serializer.go [hunk 2]
```diff
@@ -826,9 +826,9 @@ func (d *Deserializer) ReadVariableByteSlice(slice *[]byte, lenType SeriLengthPr
 
 	switch {
 	case maxLen > 0 && sliceLength > maxLen:
-		d.err = errProducer(ierrors.Wrapf(ErrDeserializationLengthInvalid, "denoted %d bytes, max allowed %d ", sliceLength, maxLen))
+		d.err = errProducer(ierrors.Join(ErrDeserializationLengthInvalid, ierrors.Wrapf(ErrDeserializationLengthMaxExceeded, "denoted %d bytes, max allowed %d ", sliceLength, maxLen)))
 	case minLen > 0 && sliceLength < minLen:
-		d.err = errProducer(ierrors.Wrapf(ErrDeserializationLengthInvalid, "denoted %d bytes, min required %d ", sliceLength, minLen))
+		d.err = errProducer(ierrors.Join(ErrDeserializationLengthInvalid, ierrors.Wrapf(ErrDeserializationLengthMinNotReached, "denoted %d bytes, min required %d ", sliceLength, minLen)))
```
Comment (piotrm50) [inline serializer/serializer.go]: Can't we wrap or join those errors in the definition of the outer one? Those will always be used together anyway. 

```
ErrDeserializationLengthMinNotReached := ierrors.Wrap(ErrDeserializationLengthInvalid, "min length not reached")
```

File: serializer/serix/decode_test.go [hunk 3]
```diff
@@ -143,7 +143,7 @@ func TestDecode_ArrayRules(t *testing.T) {
 	ts := serix.TypeSettings{}.WithLengthPrefixType(boolsLenType).WithArrayRules(rules)
 	bytesRead, err := testAPI.Decode(ctx, bytes, testObj, serix.WithValidation(), serix.WithTypeSettings(ts))
 	require.Zero(t, bytesRead)
-	assert.Contains(t, err.Error(), "min count of elements within the array not reached")
+	assert.Contains(t, err.Error(), serializer.ErrArrayValidationMinElementsNotReached.Error())
```
Comment (piotrm50) [inline serializer/serix/decode_test.go]: Can't we use `require.ErrorIs()` now, as we are comparing two errors?

File: serializer/serix/encode_test.go [hunk 4]
```diff
@@ -96,7 +96,7 @@ func TestEncode_ArrayRules(t *testing.T) {
 	ts := serix.TypeSettings{}.WithLengthPrefixType(serix.LengthPrefixTypeAsUint32).WithArrayRules(rules)
 	got, err := testAPI.Encode(ctx, testObj, serix.WithValidation(), serix.WithTypeSettings(ts))
 	require.Nil(t, got)
-	assert.Contains(t, err.Error(), "min count of elements within the array not reached")
+	assert.Contains(t, err.Error(), serializer.ErrArrayValidationMinElementsNotReached.Error())
```
Comment (piotrm50) [inline serializer/serix/encode_test.go]: Use `require.ErrorIs()`?

File: serializer/serix/map_encode_test.go [hunk 5]
```diff
@@ -30,8 +30,8 @@ func must(err error) {
 type Identifier [blake2b.Size256]byte
 
 type serializableStruct struct {
-	bytes Identifier `serix:"0"`
-	index uint64     `serix:"1"`
+	bytes Identifier `serix:"bytes"`
```
Comment (piotrm50) [inline serializer/serix/map_encode_test.go]: Do we need the JSON key in this case?

File: serializer/serix/serix.go [hunk 6]
```diff
@@ -906,67 +833,97 @@ func parseStructTag(tag string) (tagSettings, error) {
...
 		case "maxLen":
-			if len(keyValue) != 2 {
-				return tagSettings{}, ierrors.Errorf("incorrect maxLen tag format: %s", currentPart)
+			value, err := parseStructTagValueUint("maxLen", keyValue, currentPart)
+			if err != nil {
+				return tagSettings{}, err
+			}
+			settings.ts = settings.ts.WithMaxLen(value)
+
+		case "mapMaxByteSize":
```
Comment (piotrm50) [inline serializer/serix/serix.go]: Do we have all those options documented somewhere? Why do we have `mapMaxByteSize` but not `sliceMaxByteSize` or `mapValueMinLen` but not `sliceValueMinLen`? For what do we need those constraints for maps?

File: ds/set_impl.go (hunk 1 above)
Comment (muXxer) [reply ds/set_impl.go]: Change the behavior, now `""` is also fine. :thumbsup:

File: serializer/serializer.go (hunk 2 above)
Comment (muXxer) [reply serializer/serializer.go]: Done!

File: serializer/serix/map_encode_test.go (hunk 5 above)
Comment (muXxer) [reply serializer/serix/map_encode_test.go]: Removed it everywhere now.

File: serializer/serix/serix.go (hunk 6 above)
Comment (muXxer) [reply serializer/serix/serix.go]: Added package description :thumbsup:

**Review by alexsporn — APPROVED** [review]

**Pull request merged**